roles/ceph-config/tasks/main.yml

```diff
@@ -9,6 +9,24 @@
       group: "ceph"
       mode: "0755"
 
+  - name: count number of osds for non-lvm scenario
+    set_fact:
+      num_osds: "{{ devices | length }}"
+    when:
+      - devices|length > 0 and (osd_scenario == 'collocated' or osd_scenario == 'non-collocated')
+
+  - name: count number of osds for lvm scenario
+    set_fact:
+      num_osds: "{{ lvm_volumes | length }}"
+    when:
+      - lvm_volumes|length > 0 and osd_scenario == 'lvm'
+
+  - name: count number of osds for lvm-batch scenario
+    set_fact:
+      num_osds: "{{ (ceph-volume lvm batch --report devices | from_json()).osds | length }}"
```

Comment:
the ``ceph-volume lvm batch --report`` will need to be run in a separate task using the command module, with the output from the command being registered into a variable. Then that variable can be used here to set the ``num_osds`` fact.